mantidproject/mantid
# Minor clean up of documentation index page

I want to close issue #21879. This branch is the beginning of work to clean up and clarify the documentation. The Sphinx documentation has been updated to:

* Re-order the training section
	* Links to basic training and introduction
	* Has a brief description of each training link in the `Training` page
* Fitting 
	* `Fit functions` and `Fitting Minimizers` together under a single `Fitting` heading
	* Brief description and a link to the discussion of minimizers
* Change `Gallery` to `Matplotlib Plot Gallery`

**To test:**

Build new documentation and make sure that:

* Links to `Training` sections all work
* Links to `Fitting Functions` and `Fitting Minimizers` work
* Brief descriptions at the top of `Training` and `Fitting` make sense

**Release Notes** 

This does not need to be included in the release notes.
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
